internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,324 @@
+package qumulo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2024-05-01/subnets"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/qumulostorage/2024-06-19/filesystems"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/qumulo/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = FileSystemResource{}
+
+type FileSystemResource struct{}
+
+func (r FileSystemResource) ModelObject() interface{} {
+	return &FileSystemResourceSchema{}
+}
+
+type FileSystemResourceSchema struct {
+	AdminPassword     string            `tfschema:"admin_password"`
+	Location          string            `tfschema:"location"`
+	Name              string            `tfschema:"name"`
+	OfferId           string            `tfschema:"offer_id"`
+	PlanId            string            `tfschema:"plan_id"`
+	PublisherId       string            `tfschema:"publisher_id"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	StorageSku        string            `tfschema:"storage_sku"`
+	SubnetId          string            `tfschema:"subnet_id"`
+	Tags              map[string]string `tfschema:"tags"`
+	Email             string            `tfschema:"email"`
+	Zone              string            `tfschema:"zone"`
+}
+
+func (r FileSystemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return filesystems.ValidateFileSystemID
+}
+
+func (r FileSystemResource) ResourceType() string {
+	return "azurerm_qumulo_file_system"
+}
+
+func (r FileSystemResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validate.FileSystemName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"admin_password": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Sensitive:    true,
+			ValidateFunc: validate.ValidatePasswordComplexity,
+		},
+
+		"email": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
could we validate the email address, should be able to use the net/mail package `mail.ParseAddress(email)`

Reply:
added